gm4_lively_lily_pads/data/gm4_lively_lily_pads/function/mechanics/right_click_detection/process.mcfunction

```diff
@@ -0,0 +1,23 @@
+# processes the "right click detection" interaction so that it is despawned when unneeded
+# @s = gm4_llp_placement_rcd interaction 
+# at @s
+# run from mechanics/right_click_detection/loop
+
+# visuals
+particle scrape ~ ~.1 ~ 0 0 0 0 1
+
+# refresh timer
+scoreboard players set $timer gm4_llp.data 10
+
+# kill rcd if uneeded
+execute unless entity @a[tag=gm4_llp_holding_item,distance=..6] run return run kill @s
```

Comment:
One of these players being nearby is enough, so we can terminate the entity lookup after finding one.
```suggestion
execute unless entity @p[tag=gm4_llp_holding_item,distance=..6] run return run kill @s
```